### PR TITLE
Http uri spaces 2881 v17

### PR DESCRIPTION
**Pull request by catenacyber**
Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket:
https://redmine.openinfosecfoundation.org/issues/2881

Describe changes:
- configures libhtp to allow spaces in URIs

Modifies #8509 with fixing `DetectHttpProtocolValidateCallback` to use new s->init_data->buffer_index like others validators have been updated

suricata-verify-pr: 1004

https://github.com/OISF/suricata-verify/pull/1004

### CODE REVIEW

File: doc/userguide/upgrade.rst [hunk 1]
```diff
@@ -39,6 +39,9 @@ Major changes
 - IPS users: by default various new "exception policies" are set to DROP
   traffic. Please see :ref:`Exception Policies <exception policies>` for details
   on the settings and their scope.
+- Spaces are accepted in HTTP1 URIs instead of in the protocol version. That is:
```
Comment (jlucovsky) [inline doc/userguide/upgrade.rst:42]: Suggest rewording this for clarification purposes.

**Comment by suricata-qa** [comment]
Information:
 

ERROR: QA failed on SURI_TLPR1_alerts_cmp.
 | field | baseline | test | % |
|--|--|--|--| 
| | SURI_TLPR1_stats_chk | | |
| .flow.memuse | 526058088 | 1342676504 | 255.23% | 

Pipeline 14046
